src/App.tsx

```diff
@@ -34,54 +56,73 @@ function App() {
+              const createIx = SystemProgram.createAccount({
+                fromPubkey: userPublicKey,
+                newAccountPubkey: wrappedSolAccount.publicKey,
+                lamports: 10_000_000,
+                space: 165,
+                programId: TOKEN_PROGRAM_ID
+              })
 
-            const createIx = SystemProgram.createAccount({
-              fromPubkey: userPublicKey,
-              newAccountPubkey: wrappedSolAccount.publicKey,
-              lamports: 10_000_000,
-              space: 165,
-              programId: TOKEN_PROGRAM_ID
-            })
+              const initIx = Token.createInitAccountInstruction(
+                TOKEN_PROGRAM_ID,
+                NATIVE_MINT,
+                wrappedSolAccount.publicKey,
+                userPublicKey
+              )
 
-            const initIx = Token.createInitAccountInstruction(
-              TOKEN_PROGRAM_ID,
-              NATIVE_MINT,
-              wrappedSolAccount.publicKey,
-              userPublicKey
-            )
-
-            const approveIx = Token.createApproveInstruction(
-              TOKEN_PROGRAM_ID,
-              wrappedSolAccount.publicKey,
-              new PublicKey('147oKbjwGDHEthw7sRKNrzYiRiGqYksk1ravTMFkpAnv'),
-              userPublicKey,
-              [],
-              5000000
-            )
+              const approveIx = Token.createApproveInstruction(
+                TOKEN_PROGRAM_ID,
+                wrappedSolAccount.publicKey,
+                new PublicKey('147oKbjwGDHEthw7sRKNrzYiRiGqYksk1ravTMFkpAnv'),
+                userPublicKey,
+                [],
+                5000000
+              )
 
-            const ix = SystemProgram.transfer({
-              fromPubkey: userPublicKey,
-              lamports: 1_000_000,
-              toPubkey: new PublicKey('147oKbjwGDHEthw7sRKNrzYiRiGqYksk1ravTMFkpAnv')
-            })
-            const tx = new SolanaTx()
-            tx.add(createIx).add(initIx).add(approveIx)
-            const a = await connection.getLatestBlockhash()
-            tx.recentBlockhash = a.blockhash
-            tx.feePayer = userPublicKey
-            const signedTx = await NightlySolana.signTransaction(tx)
-            signedTx.partialSign(wrappedSolAccount)
-            const id = await connection.sendRawTransaction(signedTx.serialize())
-            console.log(id)
+              const tx = new SolanaTx()
+
+              tx.add(createIx).add(initIx).add(approveIx)
+
+              const a = await connection.getLatestBlockhash()
+
+              tx.recentBlockhash = a.blockhash
+              tx.feePayer = userPublicKey
+
+              if (NightlyConnectSolana.connected) {
+                const signedTx = await NightlyConnectSolana.signTransaction(tx)
+                signedTx.partialSign(wrappedSolAccount)
+                const id = await connection.sendRawTransaction(signedTx.serialize())
+                console.log(id)
+              } else {
```

Comment:
why there is the same code in 'if' and 'else' blocks? this happens multiple times here

Reply:
there is different in method to sign transaction; NightlySolana is for extensiom, NightlyConnectSolana is for mobile